Sources/Amplitude/Storages/PersistentStorage.swift

```diff
@@ -238,23 +238,36 @@ extension PersistentStorage {
         return result
     }
 
+    private func isSandboxEnabled() -> Bool {
+        let environment = ProcessInfo.processInfo.environment
+        return environment["APP_SANDBOX_CONTAINER_ID"] != nil
+    }
+
     internal func getEventsStorageDirectory(createDirectory: Bool = true) -> URL {
-        // tvOS doesn't have access to document
-        // macOS /Documents dir might be synced with iCloud
-        #if os(tvOS) || os(macOS)
-            let searchPathDirectory = FileManager.SearchPathDirectory.cachesDirectory
+        let searchPathDirectory = FileManager.SearchPathDirectory.applicationSupportDirectory
+
+        // Make sure Amplitude data is sandboxed per app
+        #if os(iOS)
+            // iOS is sandboxed by default
+            let appPath = ""
         #else
-            let searchPathDirectory = FileManager.SearchPathDirectory.documentDirectory
+            // macOS/tvOS are not sandboxed automatically
```

Comment:
The comment indicates this else clause is meant for macOS/tvOS, but it will catch all other platforms. The comment also states that tvOS is not sandboxed.

Maybe it's safe to remove platform-specific handling now that you explicitly check for a sandbox being enabled, right? The platform checking seems redundant here.

Reply:
Hi @theolampert, thanks for the review. Unfortunately this check doesn't work in iOS, hence the platform checking.

Reply:
@justin-fiedler 

I see, but it should probably still be reversed, otherwise your helper is going to return false for environments like watchOS and tvOS.

It should probably be:
```swift
#if os(macOS)
    let environment = ProcessInfo.processInfo.environment
    return environment["APP_SANDBOX_CONTAINER_ID"] != nil
#else
    return true
#endif
```